test/letters/components/Address.unit.spec.jsx

```diff
@@ -35,10 +37,22 @@ describe('<Address>', () => {
     });
 
     const component = ReactTestUtils.renderIntoDocument(<Address {...props}/>);
-    const cityInput = ReactTestUtils.scryRenderedDOMComponentsWithTag(component, 'input')[3];
-    const stateInput = ReactTestUtils.scryRenderedDOMComponentsWithTag(component, 'select')[1];
+    const form = getFormDOM(component);
 
-    expect(cityInput.value).to.contain('APO');
-    expect(stateInput.value).to.contain('AE');
+    expect(form.getElement('input[name="city"]').value).to.contain('APO');
+    expect(form.getElement('select[name="state"]').value).to.contain('AE');
+  });
+
+  it('should update the redux store when fields are updated', () => {
+    const updateSpy = sinon.spy();
+    const props = _.merge({}, defaultProps, { onUserInput: updateSpy });
+    const component = ReactTestUtils.renderIntoDocument(<Address {...props}/>);
+    const form = getFormDOM(component);
+
+    // NOTE: All address lines are currently named "address", but querySelector just picks the first one
+    form.fillData('input[name="address"]', '321 Niam');
```

Comment:
As I mentioned in slack, looking for `#errorable-text-input-12` didn't do the trick. I don't know what the _cause_ is, but when I ran mocha on only this file, it worked, but running `npm run test:unit` failed to find the element--presumably because the auto-incremented id didn't match up...but..._Why??_

Reply:
Not for this PR, but the 3 street name fields should probably have different names/labels for accessibility reasons

Reply:
Yeah, when I ran across this, I made a ticket for it [here](https://github.com/department-of-veterans-affairs/vets.gov-team/issues/4855).